auth/auth_user.py

```diff
@@ -1,17 +1,23 @@
-from pieces_os_client import UserProfile
-
+from pieces_os_client import UserProfile,AllocationsApi
+from pieces.settings import PiecesSettings
 import sublime
 
 class AuthUser:
 	output_panel = None
 	phantom_set = None
+	connect = False # should we connect to the allocation server
 
 	@classmethod
 	def on_user_callback(cls,user:UserProfile=None):
+		
 		if not user:
 			cls.login_page()
 		else:
-			cls.logout_page(user.email,user.name,user.allocation)
+			cls.logout_page(user.email,user.name,user.allocation,cls.connect)
+			if cls.connect:
+				AllocationsApi(PiecesSettings.api_client).allocations_connect_new_cloud(user)
+				cls.connect = False # Rest the connect variable
```

Comment:
This should first:

- call the allocations api to see which allocations the user has already
- if their allocation is already connected, do not create and connect a new one
- if they have an allocation, but it is not running, simply connect that allocation

only create a new allocation if the user does not already have one.